lib/1.4/dml-builtins.dml

```diff
@@ -568,21 +581,76 @@ session dml_qname_cache_t _qname_cache;
 
 // Shared parameters for attributes, registers and connects, which all produce
 // attributes. Methods get_attribute and set_attribute are abstract.
-template _conf_attribute {
-    is object;
-
-    param configuration             default "optional";
-    param persistent                default false;
-    param internal                  default (!defined documentation
-                                             && !defined desc);
-
-    param readable default this.configuration != "none";
-    param writable default this.configuration != "none";
+template _conf_attribute is object {
+    shared method set_attribute(attr_value_t value) -> (set_error_t);
+    shared method get_attribute() -> (attr_value_t);
+
+    param _attr_name : const char *;
+    param _attr_name auto;
+    param _parent_obj_class : conf_class_t **;
+    param _parent_obj_class auto;
```

Comment:
Would in general be nice to comment on every `auto` why it's needed.

Reply:
I'm not sure that would contribute much. The answers would always be "we need this info for attribute softcoding, and its auto cause there's now way to retrieve it from within DML".

Reply:
Now added a general "every param below this line is for attribute softcoding", as well as a line explaining each param.